hw/mcu/stm/stm32l1xx/src/ext/Drivers/STM32L1xx_HAL_Driver/Src/stm32l1xx_hal_spi.c

```diff
@@ -171,6 +171,7 @@ static HAL_StatusTypeDef SPI_WaitOnFlagUntilTimeout(struct __SPI_HandleTypeDef *
   * @{
   */
 
+#if 0 /* Fully defined in stm32l1xx_hal_spi_ex.c */

```

Comment:
Apart from not being related with the PR, this is not OK!

1) Don't change SDK code, unless there is no way of handling the issue elsewhere
2) Don't add `#if 0`. If something should not be here, delete it (in this case maybe it would make some sense though).
3) This is handled already in `hw/mcu/stm/stm32_common/src/stm32_driver_mod_spi.c`, so maybe what you may want is to change the `#if defined(STM32L152xC)` to `#if MYNEWT_VAL(MCU_STM32L1)` in case you are using some MCU that is not a L152xC.

Reply:
I went for the #if 0 as I don't actually understand why the 'proper' hal_spi_init() function is in the _ex file and not just in the base one.
And in fact my change is now redundant (as you note) because someone else has copied the working HAL_SPI_Init() directly into the stm32_common area with the #if for the STM32L152 (although I don't see why that is 'better' as a solution than fixing it in the STM32L1xx driver source?) The STM32L152 is close enough to the STM32L151 that I can just #define the L152 one as well...
I'll back out my change as its no longer required...

Reply:
> I don't see why that is 'better' as a solution than fixing it in the STM32L1xx driver source

Because when the SDK is updated to a newer version the maintainer will have to be extra careful to not overwrite local changes.